File: app/frontend/src/app/server/python/worker.py
import sys
import json

def process_nations(batch):
    for nation in batch['nations']:
        # Python Logic: More focused on social/political flux
        nation['population'] *= 1.000015
        nation['budget'] += nation['daily_income'] * 0.92
        
        # Add some random social noise
        import random
        nation['stability'] += random.uniform(-0.1, 0.1)
        nation['stability'] = max(20, min(100, nation['stability']))
        
    return batch

if __name__ == "__main__":
    print("Python Social Logic: ACTIVE (Managing Cluster B)")
    sys.stdout.flush()
    for line in sys.stdin:
        try:
            batch = json.loads(line)
            processed = process_nations(batch)
            print(json.dumps(processed))
            sys.stdout.flush()
        except Exception as e:
            # Silent fail for performance
            pass

File: app/frontend/src/app/server/rust/worker.rs
use std::io::{self, BufRead};
use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize, Debug, Clone)]
struct NationState {
    name: String,
    population: f64,
    budget: f64,
    happiness: f64,
    stability: f64,
    daily_income: f64,
}

#[derive(Serialize, Deserialize, Debug)]
struct SimulationBatch {
    nations: Vec<NationState>,
}

fn main() {
    println!("Rust Simulation Engine: STANDBY (Ready for 100 Nations)");
    println!("HEARTBEAT_READY"); // Signal to Go that Time Master is active
    
    let stdin = io::stdin();
    for line in stdin.lock().lines() {
        if let Ok(line_str) = line {
            if let Ok(mut batch) = serde_json::from_str::<SimulationBatch>(&line_str) {
                // Process each nation in the batch (High Performance Rust Loop)
                for nation in &mut batch.nations {
                    nation.population *= 1.00002;
                    nation.budget += nation.daily_income * 0.95;
                    
                    if nation.happiness < 50.0 {
                        nation.happiness += 0.05;
                    } else {
                        nation.happiness -= 0.02;
                    }
                }
                
                if let Ok(output) = serde_json::to_string(&batch) {
                    println!("{}", output);
                }
            }
        }
    }
}
